src/main/c/VipsImage.c

```diff
@@ -315,10 +315,17 @@ Java_com_criteo_vips_VipsImage_writeToArrayNative(JNIEnv *env, jobject obj, jstr
     size_t result_length = 0;
     int status = 0;
 
-    if (quality < 0)
-        status = vips_image_write_to_buffer(im, ext, &buffer, &result_length, "strip", strip, NULL);
-    else
-        status = vips_image_write_to_buffer(im, ext, &buffer, &result_length, "strip", strip, "Q", quality, NULL);
+    if (strcmp(ext, ".avif") == 0) {
```

Comment:
Curious why we need to do that here. I would expect the "write to buffer" functions to be identical no matter the format.

Reply:
I'm double checking if I still need it now that I fixed the issue with libheif. I'll let you know.

Reply:
Confirmed, that fixes the following errors:
```
[ERROR] Tests run: 60, Failures: 4, Errors: 0, Skipped: 3, Time elapsed: 147.215 s <<< FAILURE! - in com.criteo.vips.VipsImageTest
[ERROR] TestSimplePipelineShouldNotThrow(com.criteo.vips.VipsImageTest)  Time elapsed: 0.316 s  <<< FAILURE!
org.junit.experimental.theories.internal.ParameterizedAssertionError: TestSimplePipelineShouldNotThrow("in_vips.jpg" <from Filenames[0]>, "AVIF" <from AVIF>)
	at com.criteo.vips.VipsImageTest.TestSimplePipelineShouldNotThrow(VipsImageTest.java:547)

[ERROR] TestWriteFromDirectByteBufferShouldNotThrows(com.criteo.vips.VipsImageTest)  Time elapsed: 3.752 s  <<< FAILURE!
org.junit.experimental.theories.internal.ParameterizedAssertionError: TestWriteFromDirectByteBufferShouldNotThrows("in_vips.jpg" <from Filenames[0]>, "AVIF" <from AVIF>, "true" <from true>)
	at com.criteo.vips.VipsImageTest.TestWriteFromDirectByteBufferShouldNotThrows(VipsImageTest.java:145)

[ERROR] TestShouldWriteToArrayHasCorrectHeaderSignature(com.criteo.vips.VipsImageTest)  Time elapsed: 1.931 s  <<< FAILURE!
org.junit.experimental.theories.internal.ParameterizedAssertionError: TestShouldWriteToArrayHasCorrectHeaderSignature("in_vips.jpg" <from Filenames[0]>, "AVIF" <from AVIF>)
	at com.criteo.vips.VipsImageTest.TestShouldWriteToArrayHasCorrectHeaderSignature(VipsImageTest.java:502)

[ERROR] TestWriteFromByteArrayShouldNotThrows(com.criteo.vips.VipsImageTest)  Time elapsed: 3.67 s  <<< FAILURE!
org.junit.experimental.theories.internal.ParameterizedAssertionError: TestWriteFromByteArrayShouldNotThrows("in_vips.jpg" <from Filenames[0]>, "AVIF" <from AVIF>, "true" <from true>)
	at com.criteo.vips.VipsImageTest.TestWriteFromByteArrayShouldNotThrows(VipsImageTest.java:157)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   VipsImageTest.TestShouldWriteToArrayHasCorrectHeaderSignature TestShouldWriteToArrayHasCorrectHeaderSignature("in_vips.jpg" <from Filenames[0]>, "AVIF" <from AVIF>)
[ERROR]   VipsImageTest.TestSimplePipelineShouldNotThrow TestSimplePipelineShouldNotThrow("in_vips.jpg" <from Filenames[0]>, "AVIF" <from AVIF>)
[ERROR]   VipsImageTest.TestWriteFromByteArrayShouldNotThrows TestWriteFromByteArrayShouldNotThrows("in_vips.jpg" <from Filenames[0]>, "AVIF" <from AVIF>, "true" <from true>)
[ERROR]   VipsImageTest.TestWriteFromDirectByteBufferShouldNotThrows TestWriteFromDirectByteBufferShouldNotThrows("in_vips.jpg" <from Filenames[0]>, "AVIF" <from AVIF>, "true" <from true>)
[INFO] 
[ERROR] Tests run: 65, Failures: 4, Errors: 0, Skipped: 3
[INFO] 
```

https://github.com/libvips/libvips/blob/f1d2d4030ef0a69e67731e4102ea7d02661f1ed2/libvips/iofuncs/image.c#L2731

Reply:
Thanks for the link. I understand why libvips does that, to allow optional dependencies. It feel weird to have such a check here. I'll have to understand better how it works first.

Reply:
If you look at our calling code in [ImageVipsWrapper.scala](https://codesearch.criteois.com/gerrit-web.prod.crto.in/mediaproxy/imageproxy/-/blob/imageproxy-imageprocessor/src/main/scala/com/criteo/imageproxy/imageprocessor/native/impl/ImageVipsWrapper.scala), we explicitly call the native function based on the format. In the end, I think it's ok if this function doesn't support HEIF directly, just as the underlying native function doesn't. We're trying to minimize the behavior difference when wrapping the native calls. All that matters is that you have the `writeAVIFToArray` / `writeAVIFToArrayNative` functions. WDYT @dbouron?

Reply:
Yes it's fine to me for the moment since others bindings haven't been implemented. Moreover, we have a generic writeToArray() method which handles every supported formats